### PR TITLE
Implementing ListDatabaseSchemas

### PR DESCRIPTION
**Pull request by NamanMahor**
INSERT DESCRIPTION HERE

**Checklist:**
- [ ] Covered by tests
- [ ] Ran it and it works as intended
- [ ] Reviewed the diff before requesting a review
- [ ] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [ ] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [ ] Intend to cherry-pick into the release branch
- [ ] I'm proud of this work!

### CODE REVIEW

**Review by k-anshul — CHANGES_REQUESTED** [review]
I think we should not query across all databases by making changes to the connection string.
1. Since postgres has a notion of database isolation then we must honour that and not change connection string.
2. The rest of the code will still use the default database so it will not work if user chooses table from other database.

We must restrict to the database provided in the connection string.

**Comment by NamanMahor** [comment]
> I think we should not query across all databases by making changes to the connection string.
> 
> 1. Since postgres has a notion of database isolation then we must honour that and not change connection string.
> 2. The rest of the code will still use the default database so it will not work if user chooses table from other database.
> 
> We must restrict to the database provided in the connection string.

I think we can have a config like `allow_cross_database_metadata` which is false by default. it useful to see the databases user has access to. 
e.g in case of clickhouse user can do 
`  sql: "SELECT * FROM postgresql('{{.env.connector.postgres.ip}}:5432', 'dbname', 'tablename', 'user', 'pass')"`
user can se different db in different source. 

Edit: 
Discussed with Eric too. now only query the current database.

File: runtime/drivers/postgres/postgres.go [hunk 1]
```diff
@@ -208,13 +208,15 @@ func (c *connection) AsNotifier(properties map[string]any) (drivers.Notifier, er
 // getDB opens a new sqlx.DB connection using the config.
 func (c *connection) getDB() (*sqlx.DB, error) {
 	conf := &ConfigProperties{}
-	if err := mapstructure.WeakDecode(c.config, conf); err != nil {
+	var err error
```
Comment (k-anshul) [inline runtime/drivers/postgres/postgres.go]: Nit: No need to initialize separately ?

File: runtime/drivers/postgres/information_schema.go [hunk 2]
```diff
@@ -7,13 +7,107 @@ import (
 )
 
 func (c *connection) ListDatabaseSchemas(ctx context.Context) ([]*drivers.DatabaseSchemaInfo, error) {
-	return nil, nil
+	q := `
+	SELECT 
+	    current_database() AS database_name,
```
Comment (k-anshul) [inline runtime/drivers/postgres/information_schema.go]: Nit: Formatting
